File: leetsync/3621-minimum-operations-to-make-array-values-equal-to-k/minimum-operations-to-make-array-values-equal-to-k.java
class Solution {
    public int minOperations(int[] nums, int k) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (num < k) {
                return -1;
            } else if (num > k) {
                set.add(num);
            }
        }
        return set.size();
    }
}

File: leetsync/3621-minimum-operations-to-make-array-values-equal-to-k/minimum-operations-to-make-array-values-equal-to-k.go
func minOperations(nums []int, k int) int {
	set := make(map[int]struct{})
	for _, num := range nums {
		if num < k {
			return -1
		} else if num > k {
			set[num] = struct{}{}
		}
	}
	return len(set)
}